simulator/modules/fetch/fetch.cpp

```diff
@@ -67,15 +67,27 @@ Target Fetch<FuncInstr>::get_target( Cycle cycle)
                                                                 rp_flush_target_from_decode->read( cycle) : Target();
     const Target branch_target   = rp_target->is_ready( cycle) ? rp_target->read( cycle) : Target();
 
+    if (prefetch_enabled && prefetch_method == "wrong_path")
+    {
+        is_wrong_path = false;
+    }
+
     /* Multiplexing */
     if ( external_target.valid)
         return external_target;
 
-    if( flushed_target.valid)
+    if ( flushed_target.valid)
         return flushed_target;
 
-    if( flushed_target_from_decode.valid)
+    if ( flushed_target_from_decode.valid)
+    {
+        if (prefetch_enabled && prefetch_method == "wrong_path") // prefetch wrong path if prefetch enabled and defined
+        {
+            is_wrong_path = true;
+            tags->write( flushed_target_from_decode.address);
```

Comment:
If I understand correctly. this is not precise, as cache miss latency is not accounted here.
Also, what would happen if a line is already presented in the instruction cache?

Reply:
I added a check if the line is present in the cache or not. I think writing to cache is in the right place because the prefetch method tries to prevent cache misses. But if I am wrong about this, please explain to me.